providers/presto/src/airflow/providers/presto/hooks/presto.py

```diff
@@ -150,6 +150,35 @@ def get_conn(self) -> Connection:
 
         return presto_conn
 
+    @property
+    def sqlalchemy_url(self) -> URL:
+        """Return a `sqlalchemy.engine.URL` object constructed from the connection."""
+        conn = self.get_connection(self.get_conn_id())
+        extra = conn.extra_dejson or {}
+        catalog = extra.get("catalog", "hive")
+        schema = conn.schema
+
+        query = {"protocol": extra.get("protocol", "http"), "source": extra.get("source", "airflow")}
+
+        if schema:
+            query["schema"] = schema
+
+        url_query_params = {k: v for k, v in query.items() if v is not None}
+        host = str(conn.host) if conn.host else "test_host"
+        return URL.create(
+            drivername="presto",
+            username=conn.login or "",
+            password=conn.password or "",
+            host=host,
+            port=conn.port,
+            database=catalog,
+            query=url_query_params,
+        )
```

Comment:
```suggestion
        conn = self.get_connection(self.get_conn_id())
        extra = conn.extra_dejson or {}

        query = {
            "protocol": extra.get("protocol", "http"),
            "source": extra.get("source", "airflow"),
            "schema": conn.schema or None
        }
        url_query_params = {k: v for k, v in query.items() if v is not None}

        return URL.create(
            drivername="presto",
            username=conn.login or "",
            password=conn.password or "",
            host=str(conn.host) if conn.host else "test_host",
            port=conn.port,
            database=extra.get("catalog", "hive"),
            query=url_query_params,
        )
```


would like to confirm what is this "test_host"? is it preserved by Presto?

Reply:
Thanks for the review, @Lee-W.
That was my mistake — while testing, I unintentionally added a fallback that shouldn’t be there. Presto itself doesn’t provide any fallback for host, so in production the code should raise an error if it’s missing. I’ll remove the test fallbacks to make the hook safe and correct.